### PR TITLE
feat(tentacle): add request_session_messages_range endpoint

### PR DESCRIPTION
**Pull request by corelli18512**
## What

Adds a third session-message fetch endpoint to tentacle:
- existing **`request_session_ turn-aligned, pagination by `beforeSeq`, current iOS history fetchmessages`** 
- existing **`request_session_ deprecated but still used by web's `MessageProvider.fetchRange` for range queries (keeping for now)replay`** 
- new **`request_session_messages_ exact `(fromSeq, toSeq)` inclusive interval fetch (this PR)range`** 

## Why

Two consumers need exact-range fetches that neither existing endpoint serves cleanly:

1. **iOS gap recovery (motivating case).** When a push arrives with `seq > store.lastSeq + 1`, we today have no compensating  we either request the latest turn (loses older missing seqs) or refuse to insert (creates a hole). The new endpoint lets the arm fetch exactly the missing `(store.lastSeq + 1, push.seq - 1)` interval, no turn realignment, no over-fetch.fetch 
2. **Web range-query migration target.** `MessageProvider.fetchRange` currently leans on `request_session_replay`. Once this lands, web can migrate to a non-deprecated, well-defined endpoint and we can finally retire `request_session_replay`. (Migration happens in a follow-up PR.)

## Protocol

```ts
// @kraki/protocol
export const SESSION_MESSAGES_RANGE_MAX_COUNT = 500;

interface RequestSessionMessagesRangeMessage {
  type: 'request_session_messages_range';
  payload: { sessionId; fromSeq; toSeq }; // both inclusive
}

interface SessionMessagesRangeBatchMessage {
  type: 'session_messages_range_batch';
  payload: {
    sessionId;
    messages;     // PERSISTENT_TYPES only, each with .seq
    firstSeq;     // 0 when messages empty
    lastSeq;      // 0 when messages empty
    truncated;    // true iff server hit the 500-row cap
  };
}
```

Design decisions worth flagging (all settled in discussion):
- **Interval-based** `(fromSeq, toSeq)` not `(afterSeq,  matches the gap-recovery use case where the caller already knows both endpoints.limit)` 
- **Client-driven pagination.** No `pageSize` request parameter. The 500-row cap is a defensive backstop; iOS will hard-code a page size of 200 client-side and iterate.
- **`truncated` semantics.** When the requested range exceeds the cap, the handler keeps the **newer** end and sets `truncated: true`, so the caller's next page request shifts toward older seqs naturally.
- **No `coveredFromSeq` ambiguity.** Per-session seqs are strictly contiguous (`appendMessage` does `meta.lastSeq + 1`) and only `PERSISTENT_TYPES` ever get a seq (gated at `relay-client.ts:1780`). So `firstSeq`/`lastSeq` of the returned messages always exactly describe what was  "truncated && empty" is impossible.scanned 

## Robustness contract

The handler is defensive about  never trust the consumer:input 

| Input | Behavior |
|---|---|
| `fromSeq < 1` | clamp to 1 |
| `toSeq > headSeq` | clamp to `headSeq` |
| `fromSeq > toSeq` (post-clamp) | empty batch, `truncated: false` |
| `hi - lo + 1 > 500` | keep newer end, `truncated: true` |
| unknown sessionId | empty batch |
| NaN / non-finite | empty batch |
| transient type with a seq (older logs) | filtered out |
| fractional seq | `Math.floor` |
| per-line JSON parse error | logged, skipped |

## Tests

12 new unit tests in `relay-client.test.ts` cover every row above plus the boundary case `range == 500`. All 515 tentacle tests pass (was 503).

## Out of scope (follow-up PRs)

- **iOS gap-recovery integration** in `MessageProvider. uses `RANGE_PAGE = 200`, threshold to decide range-fetch vs `requestLatest`.observeLiveMessageSeq` 
 `request_session_messages_range`.
- **Retiring `request_session_replay`** once both arms have cut over.

### CODE REVIEW

File: packages/protocol/src/messages.ts [hunk 1]
```diff
@@ -378,6 +378,68 @@ export interface SessionMessagesBatchMessage extends BaseEnvelope {
   };
 }
 
+// ── Range-based message fetch ───────────────────────
+
+/** Server-side hard cap on the number of messages returned by a single
+ *  `request_session_messages_range` reply. Defensive backstop — clients
+ *  should chunk their own requests well below this. When the server
+ *  truncates, the response's `truncated` flag is set. */
+export const SESSION_MESSAGES_RANGE_MAX_COUNT = 500;
```
Comment (corelli18512) [inline packages/protocol/src/messages.ts]: this should not be in protocol

Comment (corelli18512) [reply packages/protocol/src/messages.ts]: Good  moved to RelayClient.RANGE_MAX_COUNT (private static, tentacle-only). Clients learn about server truncation only via the truncated flag in the response, which is the real contract. Fixed in afd7dadf.

Comment (corelli18512) [reply packages/protocol/src/messages.ts]: Moved to RelayClient.RANGE_MAX_COUNT (private static, tentacle-only). Clients learn about server truncation only via the truncated flag in the response, which is the real contract. Fixed in afd7dadf.
